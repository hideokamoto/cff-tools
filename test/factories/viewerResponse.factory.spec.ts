import {
    TestResponseEventFactory,
} from '../../src/factories/viewerResponse.factory'

describe('factories/viewerResponse.factory.ts', () => {
    describe('TestResponseEventFactory', () => {
        it('should return default event json', () => {
            const builder = TestResponseEventFactory.create()
            expect(builder.getEventObject()).toEqual(JSON.stringify({
                version: '1.0',
                context: {
                    eventType: 'viewer-response'
                },
                viewer: {
                    ip: ''
                },
                request: {
                    method: 'GET',
                    uri: '/index.html',
                    querystring: {},
                    headers: {},
                    cookies: {}
                },
                response: {
                    statusCode: 200,
                    statusDescription: 'OK',
                    headers: {},
                    cookies: {}
                }
            }))
        })
        it('should match snapshot', () => {
            const builder = TestResponseEventFactory.create()
            builder.setViewerIP('127.0.0.0')
                .setRequestMethod('POST')
                .setRequestUri('admin/index.html')
                .setRequestQuerystring({
                    test1: {
                        value: '123'
                    },
                    test2: {
                        value: '222',
                        multiValue: [{
                            value: '222'
                        }, {
                            value: '333'
                        }]
                    }
                })
                .setRequestHeaders({
                    host: {
                        value: "www.example.com"
                    },
                    accept: {
                        value: "text/html",
                        multiValue: [
                            {
                                value: "text/html"
                            },
                            {
                                value: "application/xhtml+xml"
                            }
                        ]
                    }
                })
                .setRequestCookies({
                    id: {
                        value: "CookieIdValue"
                    },
                    loggedIn: {
                        value: "false"
                    }
                })
                .setResponseCookies({
                    "id": {
                        "value": "a3fWa",
                        "attributes": "Expires=Wed, 05 Jan 2024 07:28:00 GMT",
                        "multiValue": [
                            {
                                "value": "a3fWa",
                                "attributes": "Expires=Wed, 05 Jan 2024 07:28:00 GMT"
                            },
                            {
                                "value": "a3fWb"
                            }
                        ]
                    },
                    "loggedIn": {
                        "value": "true",
                        "attributes": "Secure; Path=/; Domain=Example.com; Expires=Wed, 05 Jan 2024 07:28:00 GMT"
                    }
                })
                .setResponseHeaders({
                    "server": {
                        "value": "CustomOriginServer"
                    },
                    "content-type": {
                        "value": "text/html; charset=UTF-8"
                    },
                    "content-length": {
                        "value": "9593"
                    }
                })
                .setResponseStatusCode(404)
                .setResponseStatusDescription('Not Found')
                
                
            expect(builder.getEvent()).toMatchSnapshot()
        })
    })
})