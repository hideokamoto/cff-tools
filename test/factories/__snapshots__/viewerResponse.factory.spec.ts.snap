// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factories/viewerResponse.factory.ts TestResponseEventFactory should match snapshot 1`] = `
Object {
  "context": Object {
    "eventType": "viewer-response",
  },
  "request": Object {
    "cookies": Object {
      "id": Object {
        "value": "CookieIdValue",
      },
      "loggedIn": Object {
        "value": "false",
      },
    },
    "headers": Object {
      "accept": Object {
        "multiValue": Array [
          Object {
            "value": "text/html",
          },
          Object {
            "value": "application/xhtml+xml",
          },
        ],
        "value": "text/html",
      },
      "host": Object {
        "value": "www.example.com",
      },
    },
    "method": "POST",
    "querystring": Object {
      "test1": Object {
        "value": "123",
      },
      "test2": Object {
        "multiValue": Array [
          Object {
            "value": "222",
          },
          Object {
            "value": "333",
          },
        ],
        "value": "222",
      },
    },
    "uri": "admin/index.html",
  },
  "response": Object {
    "cookies": Object {
      "id": Object {
        "attributes": "Expires=Wed, 05 Jan 2024 07:28:00 GMT",
        "multiValue": Array [
          Object {
            "attributes": "Expires=Wed, 05 Jan 2024 07:28:00 GMT",
            "value": "a3fWa",
          },
          Object {
            "value": "a3fWb",
          },
        ],
        "value": "a3fWa",
      },
      "loggedIn": Object {
        "attributes": "Secure; Path=/; Domain=Example.com; Expires=Wed, 05 Jan 2024 07:28:00 GMT",
        "value": "true",
      },
    },
    "headers": Object {
      "content-length": Object {
        "value": "9593",
      },
      "content-type": Object {
        "value": "text/html; charset=UTF-8",
      },
      "server": Object {
        "value": "CustomOriginServer",
      },
    },
    "statusCode": 404,
    "statusDescription": "Not Found",
  },
  "version": "1.0",
  "viewer": Object {
    "ip": "127.0.0.0",
  },
}
`;
